exchanges/exchange.go

```diff
@@ -1266,3 +1267,38 @@ func (u URL) String() string {
 func (e *Base) UpdateOrderExecutionLimits(a asset.Item) error {
 	return common.ErrNotYetImplemented
 }
+
+// AssetWebsocketFunctionality defines the availability of websocket
+// functionality to the specific asset type. TODO: Deprecate as this is a temp
+// item to address certain limitations quickly.
+type AssetWebsocketFunctionality struct {
+	NoFunctionality map[asset.Item]bool
+	m               sync.RWMutex
+}
+
+// SetAssetWebsocketFunctionalityOff forces websocket functionality for the
+// supplied asset item to off. In the case that websocket functionality has not
+// yet been implemented for that specific asset type.This is a base method to
+// check availabilty of asset type
+func (b *Base) SetAssetWebsocketFunctionalityOff(aType asset.Item) error {
+	if !b.SupportsAsset(aType) {
+		return fmt.Errorf("%s %w this should be called after asset type initialisation",
+			aType,
+			asset.ErrNotSupported)
```

Comment:
I don't think it needs the flavour text since it may just be unsupported. Just `("%w %s", asset.ErrNotSupported, aType)`